### ISSUE TITLE
False positive in "optional parameter listed before required parameter"?

### ISSUE THREAD

**Issue by incase** [rodjek/puppet-lint#126]
Hi.

We have this define:

``` puppet
define apt_package($configfiles=undef,$ensure='latest') {
  package { $name: 
    ensure      => $ensure,
    configfiles => $configfiles,
    provider    => retryapt,
    require     => Class['apt::update'],
  }
}
```

Running puppet-lint on it results in:

WARNING: optional parameter listed before required parameter on line 3
WARNING: optional parameter listed before required parameter on line 4
WARNING: optional parameter listed before required parameter on line 3
WARNING: optional parameter listed before required parameter on line 3
WARNING: optional parameter listed before required parameter on line 4
WARNING: optional parameter listed before required parameter on line 3
WARNING: optional parameter listed before required parameter on line 4
WARNING: optional parameter listed before required parameter on line 3
WARNING: optional parameter listed before required parameter on line 3
WARNING: optional parameter listed before required parameter on line 4

However, I can't reproduce with a stripped down file (like posted above), only with the full file. I will send you the complete file on request. It might contain non-public stuff, so I don't want to publish it completely here. I'm trying to strip it to the minimum required to reproduce and see if I can post that.
